src/rgw/rgw_perf_counters.h

```diff
@@ -4,25 +4,26 @@
 #pragma once
 
 #include "include/common_fwd.h"
+#include "rgw_common.h"
+#include "rgw_sal.h"
+#include "common/perf_counters_cache.h"
+#include "common/perf_counters_key.h"
 
 extern PerfCounters *perfcounter;
+extern ceph::perf_counters::PerfCountersCache *user_counters_cache;
+extern ceph::perf_counters::PerfCountersCache *bucket_counters_cache;
+extern std::string rgw_op_counters_key;
```

Comment:
do we really need to expose these globals in the header? callers should be able to do everything with the functions in namespace `rgw::op_counters`, and shouldn't need direct access to these variables

if we add support for runtime changes to enable/disable these caches, we'll probably need to protect access with a mutex. encapsulating that access inside of `rgw::op_counters` makes it much easier to enforce that thread-safety